UM/Settings/DefinitionContainer.py

```diff
@@ -200,65 +199,60 @@ def hasProperty(self, key, property_name):
     #   data about inheritance and overrides was lost when deserialising.
     #
     #   Reimplemented from ContainerInterface
-    def serialize(self, ignored_metadata_keys: Optional[List] = None):
+    def serialize(self, ignored_metadata_keys: Optional[set] = None):
         data = { } # The data to write to a JSON file.
         data["name"] = self.getName()
         data["version"] = DefinitionContainer.Version
         data["metadata"] = self.getMetaData()
 
         # remove the keys that we want to ignore in the metadata
-        if ignored_metadata_keys:
-            for key in ignored_metadata_keys:
-                if key in data["metadata"]:
-                    del data["metadata"][key]
+        if not ignored_metadata_keys:
+            ignored_metadata_keys = set()
+        ignored_metadata_keys |= {"name", "version", "id", "container_type"}
```

Comment:
The whole idea of using a dictionary as storage for metadata is that it can be extensible from non-centralised places. As in, there is no pre-set set of keys. Otherwise we might as well just use a class or a namedtuple. But having a dictionary allows us to define additional metadata entries from anywhere, including plug-ins and such.